warehouse/query-core/src/test/java/datawave/query/jexl/visitors/IsNotNullPruningVisitorTest.java

```diff
@@ -575,6 +577,22 @@ public void testNoOpQueryPropertyMarkers() {
         test(query, query);
     }
 
+    @Test
+    public void testFutureCase_Custom() {
+        // logically, these unions are equivalent and the 'is not null' side can be pruned
+        String query = "FOO == 'bar' && (!(FOO == null) || !(FOO2 == null) || !(FOO3 == null) || !(FOO4 == null))";
+        String expected = "FOO == 'bar'";
+
+        test(query, expected);
+    }
+
+    @Test
+    public void testFutureCase_CustomTwo() {
+        // logically, these unions are equivalent and the 'is not null' side can be pruned
+        String query = "FOO == 'bar' && (!(FOO2 == null) || !(FOO4 == null))";
```

Comment:
I told you the wrong thing with this case. We can't prune is not null terms for mutually exclusive fields. A document like `{FOO=bar, FOO2=baz}` shouldn't be returned by this query, but if the 'not null' functions are pruned then the document will be returned.

Reply:
You're correct, it turns out the comment for this test case was not updated. This particular string will not do any pruning since FOO2 and FOO4 are not the fields in the EQNode clause. This query will end up doing no action when passed through the pruning visitor.

Reply:
Updated the comment here to clarify the function, as we chatted about a little in office.